app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -0,0 +1,633 @@
+        scroll_layer.add_exclusive(&self.label);
+        self.label.add_to_scene_layer(scroll_layer);
+        scroll_layer.add_exclusive(&self.divider);
+    }
+
+    /// Full height of the section including header.
+    fn height(&self, style: &Style) -> f32 {
+        let label_height = style.section.heading.height();
+        let body_height = self.content.height();
+        let next_section_offset = 17.0; // TODO[MM] this should not be needed.
+        body_height + label_height + next_section_offset
+    }
+
+    /// Set the top y position of the section.
+    fn set_base_position_y(&self, position_y: f32, style: &Style) {
+        match SECTION_HEADER_PLACEMENT {
+            SectionHeaderPlacement::Top => {
+                self.label.set_position_y(position_y - self.label.height.value() / 1.5);
+                self.divider.set_position_y(position_y);
+                let content_position_y =
+                    position_y - style.section.heading.offset - style.section.heading.height();
+                self.content.set_position_top_y(content_position_y);
+            }
+            SectionHeaderPlacement::Bottom => {
+                self.label.set_position_y(
+                    position_y - self.content.height() - self.label.height.value() / 1.5,
+                );
+                self.divider.set_position_y(
+                    position_y - self.content.height() - style.section.divider_height + 1.0,
+                );
+                self.content.set_position_top_y(position_y);
+            }
+        }
+    }
+}
+
+
+
+// ===========
+// === FRP ===
+// ===========
+
+define_endpoints_2! {
+    Input{
+        set_local_scope_section(list_view::entry::AnyModelProvider<component_group::Entry>),
+        set_favourites_section(Vec<LabeledAnyModelProvider>),
+        set_sub_modules_section(Vec<LabeledAnyModelProvider>),
+    }
+    Output{}
+}
+
+impl component::Frp<Model> for Frp {
+    fn init(
+        frp_api: &<Self as API>::Private,
+        _app: &Application,
+        model: &Model,
+        style: &StyleWatchFrp,
+    ) {
+        let network = &frp_api.network;
+
+        let (layout_update, init_layout) = Style::from_theme(network, style);
+
+        frp::extend! { network
+            model.favourites_section.content.set_content <+ frp_api.input.set_favourites_section;
+            model.local_scope_section.content.set_entries <+ frp_api.input.set_local_scope_section;
+            model.sub_modules_section.content.set_content <+ frp_api.input.set_sub_modules_section;
+
+            content_update <- any3_(
+                &frp_api.input.set_favourites_section,
+                &frp_api.input.set_local_scope_section,
+                &frp_api.input.set_sub_modules_section,
+            );
+
+            recompute_layout <- all(&content_update,&layout_update);
+
+            eval recompute_layout(((_,layout)) model.recompute_layout(layout) );
+        }
+        init_layout.emit(())
```

Comment:
Can we decrease the amount of emty lines here, please? I'm not sure if every line realy separates some bigger concept here. If you dont agree, feel free to skip this comment.